### PR TITLE
feat: Policy Cache mechanism improvements[staging]

### PR DESCRIPTION
**Pull request by sumandas0**
## Policy Cache mechanism improvements

> We are enhancing the way we build policy caches within the authorizer to improve authorization evaluation and reduce the number of calls made to Keycloak. This includes additional enhancements as well.
Instead of making numerous API calls to Keycloak, we will create custom views in Postgres tailored to our specific requirements. We will then fetch the necessary information from Heracles and use it to build the cache

More details: https://www.notion.so/atlanhq/Policy-Cache-mechanism-improvements-028fe879ed2a4604ae880a263726b2fe?pvs=4

JIRA: https://atlanhq.atlassian.net/browse/PLT-524

Test Cases: https://atlanhq.atlassian.net/browse/PLT-524?focusedCommentId=50549

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New Enhancement (adds functionality)

### Development

- [ ] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

**Review by n5nk — COMMENTED** [review]
refresh the branch with master, i see few unrelated commits.

File: client-auth/src/main/java/org/apache/atlas/auth/client/heracles/models/HeraclesUserViewRepresentation.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class HeraclesUserViewRepresentation {
+    protected  String id;
+    protected String username;
+    protected boolean enabled;
+    protected List<String> roles;
+    protected List<String> groups;
+
+    public static String sortBy = "username";
```
[the remark this answers is not in the document]
Comment (sumandas0) [reply client-auth/src/main/java/org/apache/atlas/auth/client/heracles/models/HeraclesUserViewRepresentation.java:16]: https://github.com/atlanhq/atlas-metastore/pull/2736/files#diff-1f8625fe034a41b30244ea6e2690e59a938be4e5a6e2743f0428f8346367079dR46

File: client-auth/src/main/java/org/apache/atlas/auth/client/heracles/models/HeraclesRoleViewRepresentation.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class HeraclesRoleViewRepresentation {
+    protected String id;
+    protected String name;
+    protected String realmId;
+    protected List<String> roles;
+    protected List<String> groups;
+
+    public static String sortBy = "name";
```
[the remark this answers is not in the document]
Comment (sumandas0) [reply client-auth/src/main/java/org/apache/atlas/auth/client/heracles/models/HeraclesRoleViewRepresentation.java:16]: https://github.com/atlanhq/atlas-metastore/pull/2736/files#diff-1f8625fe034a41b30244ea6e2690e59a938be4e5a6e2743f0428f8346367079dR60

[the remark this answers is not in the document]
Comment (sumandas0) [reply client-auth/src/main/java/org/apache/atlas/auth/client/heracles/models/HeraclesRoleViewRepresentation.java:16]: Using it here ^

File: common/src/main/java/org/apache/atlas/service/redis/AbstractRedisService.java [hunk 3]
```diff
@@ -100,6 +100,11 @@ Config getProdConfig() throws AtlasException {
                 .setClientName(ATLAS_METASTORE_SERVICE)
                 .setReadMode(ReadMode.MASTER_SLAVE)
                 .setCheckSentinelsList(false)
+                .setKeepAlive(true)
```
[the remark this answers is not in the document]
Comment (sumandas0) [reply common/src/main/java/org/apache/atlas/service/redis/AbstractRedisService.java]: Done

File: client-auth/src/main/java/org/apache/atlas/auth/client/heracles/models/HeraclesRoleViewRepresentation.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (nikhilbonte21) [reply client-auth/src/main/java/org/apache/atlas/auth/client/heracles/models/HeraclesRoleViewRepresentation.java:16]: rename to something like SORT_KEY_BY_NAME..

File: intg/src/main/java/org/apache/atlas/AtlasConfiguration.java [hunk 4]
```diff
@@ -108,7 +108,8 @@ public enum AtlasConfiguration {
 
     PERSONA_POLICY_ASSET_MAX_LIMIT("atlas.persona.policy.asset.maxlimit", 1000),
     ENABLE_KEYCLOAK_TOKEN_INTROSPECTION("atlas.canary.keycloak.token-introspection", false),
-    KEYCLOAK_ADMIN_CLIENT_PAGINATION_SIZE("atlas.keycloak.admin.resource-pagination-size", 1500);
+    HERACLES_CLIENT_PAGINATION_SIZE("atlas.heracles.admin.resource-pagination-size", 100),
+    HERACLES_API_SERVER_URL("atlas.heracles.api.server.url", "http://heracles-service.heracles.svc.cluster.local");
```
Comment (nikhilbonte21) [inline intg/src/main/java/org/apache/atlas/AtlasConfiguration.java:112]: rename to atlas.heracles.api.service.url

**Review by nikhilbonte21 — APPROVED** [review]

File: intg/src/main/java/org/apache/atlas/AtlasConfiguration.java (hunk 4 above)
Comment (sumandas0) [reply intg/src/main/java/org/apache/atlas/AtlasConfiguration.java]: Done

**Pull request merged**